src/kwlib.sh

```diff
@@ -525,3 +525,59 @@ function get_file_name_from_path()
 
   printf '%s\n' "${file_path##*/}"
 }
+
+# Checks if the command is being run inside a git work-tree
+#
+# @flag: How to display (or not) the command used
+#
+# Returns:
+# 0 if is inside a git work-tree root and 128 otherwise.
+function is_inside_work_tree()
+{
+  local flag="$1"
+  local cmd='git rev-parse --is-inside-work-tree &> /dev/null'
+
+  flag=${flag:-'SILENT'}
+
+  cmd_manager "$flag" "$cmd"
+}
+
+# Get all instances of a given git config with their scope
+#
+# @flag: How to display (or not) the command used
+# @scope: Scope to use
+# @config: Given configuration to get the values of
+#
+# Returns:
+# All values of the given config with their respective scopes
+function get_all_git_config()
+{
+  local flag="$1"
+  local scope="$2"
+  local config="$3"
+  local cmd="git config $scope --get-all"
+
+  flag=${flag:-'SILENT'}
+
+  cmd_manager "$flag" "$cmd $config"
+}
+
+# Get all instances of a given git config with their scope
+#
+# @flag: How to display (or not) the command used
+# @regexp: Given regular expression to find associated values
+# @scope: Scope to use
+#
+# Returns:
+# All config values that match the given regular expression
+function get_git_config_regex()
+{
+  local flag="$1"
+  local scope="$2"
```

Comment:
I think currently the scope is always passed to this function but I've been using local as the default to avoid messing the system on accident. I'll add these changes.